src/hypofuzz/provider.py

```diff
@@ -134,29 +142,71 @@ def __init__(
     ) -> None:
         super().__init__(conjecturedata)
         self.collector = collector
-        self._index = 0
-        self._replay_queue: SortedList[ReplayQueueElement] = SortedList(
-            key=lambda x: (x[0], _choices_size(x[1]))
-        )
 
-        self.random = Random()
+        # These two attributes are incremented for every input executed,
+        # regardless of origin. This tracks the *total* effort expended by this
+        # worker on this test.
+        #
+        # These attributes are written to reports and used for coverage graph
+        # display. They are NOT used for estimator state.
+        self.status_counts = StatusCounts()
         self.elapsed_time = 0.0
+        # These four attributes are incremented only for inputs generated from a
+        # mutator. This tracks the *useful* effort expended by this worker on this
+        # test.
+        #
+        # These four attributes are not written to reports. They are used for
+        # estimator state.
+        #
+        # Invariant: elapsed_time_mutated <= elapsed_time and
+        # status_counts_mutated <= status_counts.
+        self.status_counts_mutated = StatusCounts()
+        self.elapsed_time_mutated = 0.0
         self.since_new_behavior = 0
         self.since_new_fingerprint = 0
-        self.phase: Optional[Phase] = None
-        self.status_counts = StatusCounts()
-
-        self._last_timed_report: Report | None = None
-        self._last_saved_report_at = -math.inf
-        self._last_observed = -math.inf
-        self._started = False
 
+        self.random = Random()
+        self.phase: Optional[Phase] = None
         self.worker_identity: Optional[WorkerIdentity] = None
         self.database_key: Optional[bytes] = None
         self.corpus: Optional[Corpus] = None
         self.db: Optional[HypofuzzDatabase] = None
+
+        # "replay queue" is the queue for the initial load from the database,
+        # and is not used afterwards.
+        #
+        # When we start pulling from this queue, we start Phase.REPLAY. When this
+        # queue is empty, we end Phase.REPLAY.
+        #
+        # We have two queues because we want to process the entire corpus queue
+        # before any other enqueued elements, even if they get enqueued at a
+        # nominally higher priority before the corpus queue finishes.
+        #
+        # Essentially we have `ReplayPriority x [0, 1]`, which we could model
+        # by having both ${priority} and ${priority}_corpus and for each priority
+        # such that the corpus priorities are all higher, or we could model as
+        # two separate queues. This is the latter approach.
+        self._replay_queue: SortedList[QueueElement] = SortedList(
+            key=lambda x: (x[0], _choices_size(x[1]))
+        )
```

Comment:
Hmm, talk me through this reasoning?  I'd be inclined to use a single queue, on the basis that if another worker finds a failure (etc), then we likely want to replay that _before_ we finish initial load from the database - it's potentially a fairly substantial timesaver.  I also like the simplicity of having a single queue.

Reply:
I don't think it's double?  All failures can be the same priority whether from this or another process, and `STABILITY` can't come from another worker, so it's just `COVERING_LOCAL` plus `COVERING_REMOTE` I think.